web-server/routes/lectures.js

```diff
@@ -0,0 +1,32 @@
+const fetch = require('node-fetch')
+const qs = require('querystring')
+const API_KEY = process.env.YOUTUBE_API_KEY
+const URL = 'https://www.googleapis.com/youtube/v3/playlistItems'
+
+const generateApiUrl = function(pageToken) {
+  const query = qs.stringify({
+    part: 'snippet',
+    maxResults: '48',
+    playlistId: 'UU599lkzf-2haPTzDAMGEgLg',
+    key: API_KEY,
+    pageToken,
+  })
+  return `${URL}?${query}`
+}
+
+module.exports = app => {
+  app.get('/lectures(/:pageToken)?', (request, response, next) => {
+    const { pageToken } = request.params
+    const url = generateApiUrl(pageToken)
+    fetch(url)
+      .then(function(res) {
+          return res.json();
+      }).then(function(json) {
+          app.locals.nextPageToken = json.nextPageToken || ''
```

Comment:
well this is a downside of not using that router pattern. You're modifying the locals for this app instance for all time by setting values on `app.locals`. This is bad. This means we have bleedover from one request to another. Thats death for a webapp. We should always use `response.locals` within a route.

Reply:
on that thought. setting values on `response.locals` and passing locals in via the second argument to render, do the same thing. so lets do one or the other